politeiawww/api/cms/v1/v1.go

```diff
@@ -433,3 +446,37 @@ type PaymentInformation struct {
 	AmountReceived  dcrutil.Amount `json:"amountreceived"`
 	Status          PaymentStatusT `json:"status"`
 }
+
+// CMSUser represents a CMS user. It contains the standard politeiawww user
+// fields as well as CMS specific user fields.
+type CMSUser struct {
+	User               www.User        `json:"user"`
+	Domain             DomainTypeT     `json:"domain"` // Contractor domain
+	GitHubName         string          `json:"githubname"`
+	MatrixName         string          `json:"matrixname"`
+	ContractorType     ContractorTypeT `json:"contractortype"`
+	ContractorName     string          `json:"contractorname"`
+	ContractorLocation string          `json:"contractorlocation"`
+	ContractorContact  string          `json:"contractorcontact"`
+	SupervisorUserID   string          `json:"supervisoruserid"`
+}
+
+// UserDetails fetches a cms user's details by their id.
+type UserDetails struct {
+	UserID string `json:"userid"` // User id
+}
+
+// UserDetailsReply returns a cms user's details.
+type UserDetailsReply struct {
+	User *CMSUser `json:"user"`
+}
+
+// EditUser edits a user's preferences.
+type EditUser struct {
+	CMSUser   CMSUser `json:"user"`
```

Comment:
This is pretty ambiguous.  What user fields are being edited?  This implies the entire`CMSUser` object is able to be edited.

A better way to do this would be to only include the fields that are eligible to be edited by the route.